mpmp/src/model/Plot.java

```diff
@@ -105,24 +113,35 @@ public int hypothec(boolean addhyp) {
 		return hypothecValue;
 	}
 
+	/**
+	 * Search for the plot of that name.
+	 */
 	public static Plot search(String name) {
 		for (Plot p : allplots)
 			if (name.equals(p.getName()))
 				return p;
 		return null;
 	}
-	
-	public static String matches(String name) {
-		Iterator i = allplots.iterator();
+
+	/**
+	 * Find longest string in the input that matches one of the plot names.
+	 */
+	public static String matches(String s) {
 		String bestMatch = "";
-		while (i.hasNext()) {
-			Plot p = (Plot) i.next();
-			String pname = p.getName();
-			if (name.matches(pname + "(.*)") && pname.length() > bestMatch.length())
-				bestMatch = pname;
-		}
+		for(Plot p : allplots)
+			if (s.matches(p.name + "(.*)") && p.name.length() > bestMatch.length())
+				bestMatch = p.name;
```

Comment:
@leletec Two notes: first, why did you think you'd need an explicit iterator here? Second, you don't need to call `p.getName()` if you're already in `p`'s class. Encapsulation doesn't happen between objects, it happens between clases.